api/src/Database/DatabaseQueryBuilder.php

```diff
@@ -71,30 +73,62 @@ public function whereIdEquals($columnName, $value) {
      * 
      * @param string $expectedTable Table name
      */
-    public function update($expectedTable) {
+    public function update($expectedTable)
+    {
 
-        if (sizeof($this->update_values) < 1) 
+        if (sizeof($this->update_values) < 1)
             return null;
-        $set = $this->bindArray($this->update_values);
+        $set = $this->bindArrayAsList($this->update_values);
 
         $where = $this->where_columnName . "=:" . $this->addBoundVariable($this->where_value);
 
         return $this->db->pq("UPDATE {$expectedTable} SET {$set} WHERE {$where}", $this->query_bound_values);
     }
 
-    private function bindArray($values)
+    /**
+     * Run the insert if any patch parameters have been set 
+     * 
+     * @param string $expectedTable Table name
+     * @return ?int|string The auto increment number as a number unless it is greater than int in which case a string. 
+     *      0 if no auto-increment column
+     */
+    public function insert($expectedTable)
+    {
+
+        if (sizeof($this->update_values) < 1)
+            return null;
+        $values = $this->bindArrayAsInsertValues($this->update_values);
+
+        $this->db->pq("INSERT INTO {$expectedTable} {$values}", $this->query_bound_values);
+        return $this->db->id();
+    }
+
+    private function bindArrayAsList($values) : string
     {
         $bound_list_sql = "";
-        foreach ($values as $fieldname => $fieldvalue){
+        foreach ($values as $fieldname => $fieldvalue) {
             $bound_list_sql .= ", {$fieldname}=:" . $this->addBoundVariable($fieldvalue);
         }
         return substr($bound_list_sql, 2); // remove first comma
     }
 
-    private function addBoundVariable($value)
+    private function bindArrayAsInsertValues($values) : string
+    {
+        $value_names = "";
+        $value_binds = "";
+        foreach ($values as $fieldname => $fieldvalue) {
+            $value_names .= ", {$fieldname}";
+            $value_binds .= ", :" . $this->addBoundVariable($fieldvalue);
+        }
+        $value_names = substr($value_names, 2);  // remove first comma
+        $value_binds = substr($value_binds, 2);  // remove first comma
+
+        return "( {$value_names} ) VALUES ( ${value_binds} )";
```

Comment:
Should we return the value arrays rather than part of the SQL query?

Reply:
done